web/src/App.tsx

```diff
@@ -31,21 +36,34 @@ const TimeControllerWrapper = lazy(() => import('features/time/TimeControllerWra
 
 const isProduction = import.meta.env.PROD;
 
-if (isProduction) {
-  trackEvent(TrackEvent.APP_LOADED, {
-    isNative: Capacitor.isNativePlatform(),
-    platform: Capacitor.getPlatform(),
-  });
-}
+export const useInitialState = () => {
+  const setSelectedDatetimeIndex = useSetAtom(selectedDatetimeIndexAtom);
+  const setTargetDatetimeString = useSetAtom(targetDatetimeStringAtom);
+  const parsedPath = parsePath(location.pathname);
+
+  // Set initial datetime synchronously
+  useLayoutEffect(() => {
+    if (parsedPath?.datetime) {
+      const pathDate = new Date(parsedPath.datetime);
+      if (!Number.isNaN(pathDate.getTime())) {
+        setTargetDatetimeString(parsedPath.datetime);
+        setSelectedDatetimeIndex({
+          datetime: pathDate,
+          index: 24,
```

Comment:
I don't think we can set the index to 24 here, at least without additional checks. Currently this also makes it so we have 25 months and years of data 😅